src/embedded.ts

```diff
@@ -147,6 +149,26 @@ export class RoomWidgetClient extends MatrixClient {
     ) {
         super(opts);
 
+        const transportSend = this.widgetApi.transport.send.bind(this.widgetApi.transport);
```

Comment:
Why do we need this bind?

Reply:
This is used as a way to preserve the original `send` method so it can still be used by the function that replaces it.

Alternatively this could have been an arrow function, but that would require specifying a parameter list.  Using bind gets around that.

Reply:
This seems like a really nasty pattern, this looks like somewhere a subclass makes far more sense rather than overwriting class methods, what's the rationale for not doing it in a more conventional way?

Reply:
It's because the `send` methods don't belong to `WidgetApi`, but to `ITransport`, meaning that subclassing (or even a composition-based approach) would have to work harder to get to them.

Any non-`bind` approach I can think of ends up being more complicated / requires much more copied code than this.